dockerrotate/images.py

```diff
@@ -112,8 +112,19 @@ def clean_images(args):
             continue
 
         try:
+            # The simplest way to do this would be to delete by ID. However, then we
+            # encounter issues in the case where we have an image A that is tagged for
+            # removal, but that image is a parent image for image B. The desired behavior
+            # in that case is that all tags are removed for that image, but the image
+            # itself remains until B is removed.
+            #
             # force=true is required here because the image we remove might be "latest".
-            args.client.remove_image(image["Id"], force=True, noprune=False)
+
+            for repo_tag in image["RepoTags"]:
+                args.client.remove_image(repo_tag, force=True, noprune=False)
+            else:
```

Comment:
Good use of `else` on the loop. Might it be clearer that the only change is from tag(s) to id the `remove_image` didn't repeat? e.g.,
```
            # If the image has no tags (unexpected), fall back to ID.
            for identifier in image["RepoTags"] or [image["Id"]]:
                args.client.remove_image(identifier, force=True, noprune=False)
```